soroban-env-host/src/budget/model.rs

```diff
@@ -72,16 +72,18 @@ impl Display for ScaledU64 {
     }
 }
 
-impl Debug for ScaledU64 {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "Scaled({})", self.0)
+impl From<f64> for ScaledU64 {
```

Comment:
Shall we guard this with benchmark feature?

Reply:
I think it's a fairly harmless / intrinsic aspect of the type. The type represents a fixed-point fraction, it's reasonable to equip it with a (lossy but exactly-as-you-would-expect) conversion from floating-point. Not that anyone's going to _use_ that outside benchmarking, but it seems harmless to me.

Reply:
We just avoided floats in host code altogether, but I don't have a strong opinion on this either.

Reply:
I've put it behind the feature gate. 
I think it is good to have, just to be extra clear floating points aren't enabled in the regular code flow.